### ISSUE TITLE
GitHub: discard webhooks not matching autodiscoverFilter/ignore repositories completely

### ISSUE THREAD

**Issue by morremeyer** [mend/renovate-ce-ee#303]
This touches two closely related topics, which is why I'm only opening one issue for this for now. 

TL;DR: We'd like to be able to make our on-prem GitHub app public and ignore any repository that does not match the `autodiscoverFilter` completely.

The [documentation states](https://github.com/mend/renovate-on-prem/blob/be18789ff9e5e799ce3f2b0d70debf23591cfe40/docs/advanced.md#horizontal-scaling) that when running with GitLab, webhooks are ignored for repositories that do not match the `autodiscoverFilter`.
This behavior would be great to have for GitHub, too, since we have > 10 GitHub organizations that we want to run on without creating a GitHub app for every single one.

If I understand it correctly, with this behavior any repository not matching the `autodiscoverFilter` would be completely ignored.

This is the behavior we'd like to achive since this would enable us to make the GitHub app public.
With the current behavior, anyone could install a public app and while no runs would happen on the schedule, they would still occur for webhooks.

**Comment by justo-mend** [comment]
This has now been addressed in the released versions.
